src/esp/sensor/SensorFactory.h

```diff
@@ -0,0 +1,16 @@
+#include "esp/scene/SceneNode.h"
+#include "esp/sensor/CameraSensor.h"
+#include "esp/sensor/Sensor.h"
+
+namespace esp {
+class SensorFactory {
+ public:
+  virtual ~SensorFactory();
+  static sensor::SensorSuite createSensors(
+      scene::SceneNode& node,
+      const sensor::SensorSetup& sensorSetup);
+
+ private:
+  ESP_SMART_POINTERS(SensorFactory)
```

Comment:
Don't think you need this either, since we're not going to make SensorFactory objects.  
Besides, if we do and we want to reference them as smart pointers, this macro needs to be public for it to be usable.